database/src/main/java/com/firebase/ui/database/FirebaseIndexArray.java

```diff
@@ -75,39 +78,39 @@ private int getIndexForKey(String key) {
     }
 
     private boolean isKeyAtIndex(String key, int index) {
-        return index >= 0 && index < getCount() && mDataSnapshots.get(index).getKey().equals(key);
+        return index >= 0 && index < size() && mDataSnapshots.get(index).getKey().equals(key);
     }
 
     @Override
     public void onChildAdded(DataSnapshot keySnapshot, String previousChildKey) {
-        super.setOnChangedListener(null);
+        setShouldNotifyListeners(false);
```

Comment:
@samtstern Yeah, I would love to improve that too, but I'm not sure how to do so. See the original discussion: https://github.com/firebase/FirebaseUI-Android/pull/276#pullrequestreview-503792. The way I did it originally was to override the `notify...` methods and make them noops, but @MaciejCiemiega pointed out that if devs were to extend `FirebaseIndexArray`, they would expect `notify...` to actually do something. If you can think of another way to solve this problem, I would love to hear it!